spec/javascripts/app/views/notifications_view_spec.js

```diff
@@ -92,78 +72,65 @@ describe("app.views.Notifications", function(){
         it("changes the header notifications count", function() {
           var badge1 = $(".notifications-link:eq(0) .badge");
           var badge2 = $(".notifications-link:eq(1) .badge");
-          var count = parseInt(badge1.text(), 10);
 
-          this.view.updateView(this.guid, this.type, true);
-          expect(parseInt(badge1.text(), 10)).toBe(count + 1);
+          expect(parseInt(badge1.text(), 10)).toBe(this.collection.unreadCount);
+          expect(parseInt(badge2.text(), 10)).toBe(this.collection.unreadCount);
 
-          this.view.updateView(this.guid, this.type, false);
-          expect(parseInt(badge1.text(), 10)).toBe(count);
+          this.collection.unreadCount++;
+          this.view.updateView();
+          expect(parseInt(badge1.text(), 10)).toBe(this.collection.unreadCount);
 
-          this.view.updateView(this.guid, this.type, true);
-          expect(parseInt(badge2.text(), 10)).toBe(count + 1);
+          this.view.updateView();
+          expect(parseInt(badge2.text(), 10)).toBe(this.collection.unreadCount);
+        });
```

Comment:
Before we checked both increase and decrease of the unread counts. Calling `updateView` once should also be sufficient for changes to `badge1` and `badge2`, so just do "increase", `updateView`, "check badge1 and badge2", "decrease", `updateView`, "check badge1 and badge2".